### PR TITLE
Change solicitor email address

### PR DESCRIPTION
**Pull request by Gareth40343**
https://tools.hmcts.net/jira/browse/CMC-1895

- Added new CHANGE_SOLICITOR_EMAIL event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by HarryH96 — DISMISSED** [review]

**Review by sankaviv1 — APPROVED** [review]

File: ccd-definition/CaseField/CaseField.json [hunk 1]
```diff
@@ -1303,6 +1303,15 @@
     "FieldType": "Label",
     "SecurityClassification": "Public"
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "userRoles",
```
Comment (bhssamray) [inline ccd-definition/CaseField/CaseField.json:1308]: What is this used for?

File: ccd-definition/CaseField/CaseField.json [hunk 2]
```diff
@@ -1718,5 +1754,33 @@
...
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "amendApplicant1SolicitorEmailLabel",
+    "Label": "Update the claimant's legal representative's email address by replacing the existing email.",
+    "FieldType": "Label",
+    "SecurityClassification": "Public"
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "amendApplicant2SolicitorEmailLabel",
```
Comment (bhssamray) [inline ccd-definition/CaseField/CaseField.json:1781]: This doesn't seems to be used in the CaseEventtoField json.

File: e2e/fixtures/ccd/expectedEvents.js [hunk 3]
```diff
@@ -4,21 +4,26 @@ module.exports = {
   applicant_solicitor: {
```
Comment (bhssamray) [inline e2e/fixtures/ccd/expectedEvents.js:4]: Please can you create a tech-debt ticket in the backlog to add this event into the end-to-end tests that we have.

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json [hunk 4]
```diff
@@ -3014,12 +3014,30 @@
...
     ]
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseFieldID": "isApplicant1",
+    "UserRole": "caseworker-civil-solicitor",
+    "CRUD": "CRUD"
+  },
+  {
+    "CaseTypeID": "CIVIL",
```
Comment (bhssamray) [inline ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json:3024]: Is this needed and used (see a later comment around this)

File: ccd-definition/CaseEvent/User/UserEvents.json [hunk 5]
```diff
@@ -232,6 +232,24 @@
     "RetriesTimeoutAboutToStartEvent": 0,
     "RetriesTimeoutURLAboutToSubmitEvent": 0
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "CHANGE_SOLICITOR_EMAIL",
+    "Name": "Change solicitor email",
+    "Description": "Change solicitor email",
+    "DisplayOrder": 17,
+    "PreConditionState(s)": "PENDING_CASE_ISSUED;CASE_ISSUED;AWAITING_CASE_DETAILS_NOTIFICATION;AWAITING_RESPONDENT_ACKNOWLEDGEMENT;AWAITING_APPLICANT_INTENTION;CASE_DISMISSED",
```
[the remark this answers is not in the document]
Comment (Gareth40343) [reply ccd-definition/CaseEvent/User/UserEvents.json]: Not allowed to change email when case is dismissed. Updated now.

File: ccd-definition/CaseField/CaseField.json [hunk 6]
```diff
@@ -1317,13 +1326,40 @@
...
+    "CaseTypeID": "CIVIL",
+    "ID": "isApplicant1",
+    "Label": " ",
+    "FieldType": "YesOrNo",
+    "SecurityClassification": "Public",
+    "_Comments": "Temporary workaround - flag used to indicate applicant 1 for whom an event should apply.",
+    "_Category": "WORKAROUND"
+  },
+  {
+    "CaseTypeID": "CIVIL",
```
[the remark this answers is not in the document]
Comment (Gareth40343) [reply ccd-definition/CaseField/CaseField.json:1330]: Removed

File: e2e/fixtures/ccd/expectedEvents.js (hunk 3 above)
Comment (douglasrice) [reply e2e/fixtures/ccd/expectedEvents.js:4]: This shouldn't be a tech debt ticket - the tests should be updated as part of this ticket.
Please add this to to the API and e2e tests for 1v2 diff sols please

**Pull request merged**